app/models/solr_document.rb

```diff
@@ -38,4 +38,12 @@ class SolrDocument
   # and Blacklight::Document::SemanticFields#to_semantic_values
   # Recommendation: Use field names from Dublin Core
   use_extension(Blacklight::Document::DublinCore)
+
+  def location_display
```

Comment:
@nicktiberi @awead What do you think for moving this logic a component since It is mostly about the display? Also kinda thinking that it might be better to keep solr_document as lean as possible for blacklight updates 🤔

Reply:
That's an excellent point. ShelfItem can contain many SolrDocument objects, but maybe this an inappropriate use of SolrDocument. We could define a mini-presenter object, ex. `ShelfItem::Record`, serves as the thing that holds the solr document data, and put all the necessary display logic in there. It also opens up the question (in my mind) that maybe ShelfItem is also a presenter and not a model. 🤷 Anyway, @banukutlu's point is well-taken, and this logic ought to be someplace else. Happy to pair up with anyone to work this out.